CMakeLists.txt

```diff
@@ -0,0 +1,182 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+include(CMakeDependentOption)
+include(CheckSymbolExists)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_POSITION_INDEPENDENT_CODE YES)
+
+option(ENABLE_DEBUG_POSTFIX "Add \"d\" after library names for debug builds" ON)
+if(ENABLE_DEBUG_POSTFIX)
+    set(CMAKE_DEBUG_POSTFIX d)
+endif()
+
+# The API (SO) and detailed library versions for the shared library.
+set(PROJECTM_SO_VERSION "3")
+set(PROJECTM_LIB_VERSION "3.1.1")
+
+project(projectm
+        LANGUAGES C CXX
+        VERSION 3.1.13
+        )
+
+if(NOT MSVC)
+    include(EnableCFlagsIfSupported)
+    enable_cflags_if_supported(
+            -Wall
+            -Wchar-subscripts
+            -Wformat-security
+            -Wpointer-arith
+            -Wshadow
+            -Wsign-compare
+            -Wtype-limits
+    )
+endif()
+
+set(PROJECTM_BIN_DIR "bin" CACHE STRING "Executable installation directory, relative to the install prefix.")
+set(PROJECTM_DATADIR_PATH "share/projectM" CACHE STRING "Default (absolute) path to the projectM data files (presets etc.)")
+set(PROJECTM_LIB_DIR "lib" CACHE STRING "Library installation directory, relative to the install prefix.")
+set(PROJECTM_INCLUDE_DIR "include" CACHE STRING "Header installation directory, relative to the install prefix.")
+
+# Feature options, including dependencies.
+option(ENABLE_PRESETS "Build and install bundled presets" ON)
+option(ENABLE_TESTING "Build and install the projectM test suite" OFF)
+option(ENABLE_EMSCRIPTEN "Build for web with emscripten" OFF)
+cmake_dependent_option(ENABLE_SDL "Enable SDL2 support" ON "NOT ENABLE_EMSCRIPTEN;ENABLE_TESTING" ON)
+cmake_dependent_option(ENABLE_GLES "Enable OpenGL ES support" OFF "NOT ENABLE_EMSCRIPTEN" ON)
+cmake_dependent_option(ENABLE_THREADING "Enable multithreading support" ON "NOT ENABLE_EMSCRIPTEN" OFF)
+cmake_dependent_option(ENABLE_PULSEAUDIO "Build Pulseaudio-based Qt UI" OFF "ENABLE_QT;NOT ENABLE_EMSCRIPTEN;CMAKE_SYSTEM_NAME STREQUAL Linux" OFF)
+cmake_dependent_option(ENABLE_JACK "Build JACK-based Qt and SDL UIs" OFF "ENABLE_QT;NOT ENABLE_EMSCRIPTEN;CMAKE_SYSTEM_NAME STREQUAL Linux" OFF)
+cmake_dependent_option(ENABLE_LLVM "Enable LLVM JIT support" OFF "NOT ENABLE_EMSCRIPTEN" OFF)
+cmake_dependent_option(ENABLE_LIBVISUAL "Build and install the projectM libvisual plug-in" OFF "NOT ENABLE_EMSCRIPTEN;CMAKE_SYSTEM_NAME STREQUAL Linux" OFF)
+
+option(USE_SYSTEM_GLM OFF "Use system/external GLM library instead of the bundled one")
```

Comment:
Would be cool if we could detect and use system glm if available

Reply:
Sure, could just call ‘find_package(GLM)‘ without ‘REQUIRED‘ and then check the ‘GLM_FOUND‘ variable. If it's set to ‘NOTFOUND‘, use the bundled library and output a message.

Reply:
Fixed in commit aa34bd5